src/main/java/com/changer/marspictures/MarsPicturesService.java

```diff
@@ -0,0 +1,56 @@
+package com.changer.marspictures;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class MarsPicturesService {
+
+    private final NasaApiInfo nasaApiInfo;
+    private final RestTemplate restTemplate;
+
+    @Cacheable("pictures")
+    public Optional<byte[]> findLargestPicture(int sol, String camera) {
```

Comment:
you don`t need to load the whole picture in your app memory, you can just get the Pictures URI from the nasaAPI and use redirect to this URI in your controller. smth like this 
```
 ResponseEntity.status(MOVED_PERMANENTLY)
                .header(LOCATION, largestPicture.getPhotoUrl())
                .build();
```